apps/client/src/app/pages/voyage-tracker/voyage-tracker/voyage-tracker.component.ts

```diff
@@ -0,0 +1,99 @@
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { finalize, map, switchMap, tap } from 'rxjs/operators';
+import { NzModalService } from 'ng-zorro-antd/modal';
+import { IpcService } from '../../../core/electron/ipc.service';
+import { FreecompanyWorkshopFacade } from '../../../modules/freecompany-workshops/+state/freecompany-workshop.facade';
+import { LazyDataService } from '../../../core/data/lazy-data.service';
+import { BehaviorSubject, timer } from 'rxjs';
+import { TranslateService } from '@ngx-translate/core';
+import { VesselType } from '../../../modules/freecompany-workshops/model/vessel-type';
+import { Observable } from 'rxjs/Observable';
+import { Vessel } from '../../../modules/freecompany-workshops/model/vessel';
+
+@Component({
+  selector: 'app-voyage-tracker',
+  templateUrl: './voyage-tracker.component.html',
+  styleUrls: ['./voyage-tracker.component.less'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class VoyageTrackerComponent implements OnInit {
+  isLoading$ = new BehaviorSubject(true);
+  private machinaToggle = false;
+
+  public airshipMaxRank$ = new BehaviorSubject(null);
+  public submarineMaxRank$ = new BehaviorSubject(null);
+
+  public display$ = this.freecompanyWorkshopFacade.workshops$.pipe(
+    map((results) => {
+      this.isLoading$.next(true);
+      const worlds = {};
+      results.forEach((fc) => {
+        if (worlds[fc.world] === undefined) {
+          worlds[fc.world] = [];
+        }
+        worlds[fc.world].push(fc);
+      });
+      return worlds;
+    }),
+    tap(() => {
+      this.isLoading$.next(false);
+    }),
+    switchMap((results) => timer(0, 1000).pipe(
+      map(() => results)
+    )),
+    finalize(() => {
+      this.isLoading$.next(false);
+    })
+  );
+
+  constructor(private dialog: NzModalService, private ipc: IpcService,
+              private lazyData: LazyDataService, public translate: TranslateService,
+              private freecompanyWorkshopFacade: FreecompanyWorkshopFacade) {
+  }
+
+  ngOnInit(): void {
+    this.ipc.once('toggle-machina:value', (event, value) => {
+      this.machinaToggle = value;
+    });
```

Comment:
You can use `this.ipc.machinaToggle` to get the toggle, no need to ask electron for it.